lib/zhr_devs/submissions/events/solution_submitted.ex

```diff
@@ -0,0 +1,16 @@
+defmodule ZhrDevs.Submissions.Events.SolutionSubmitted do
+  @moduledoc """
+  Represents an event that is emitted when a solution is submitted.
+  """
+
+  @derive Jason.Encoder
+  import Algae
+
+  defdata do
```

Comment:
These are exactly the fields from that other file. Not a huge fan of code duplication? Doesn't matter I guess, but still that @fields from that other file is a bit of a code smell imho

Reply:
It is impossible to use a framework (and I even not talking about Commanded) and make no "code smells" at all, in my opinion. I don't see any "code smells" here. It is a part of our business rules.

Reply:
Oh, that's a very good way of thinking about it! I've never quite thought about the code smell this way, but yeah, it's a clear consequence of encoding an open system using elements of a closed system!